source/vir/src/datatype_to_air.rs

```diff
@@ -74,7 +74,8 @@ fn uses_ext_equal(ctx: &Ctx, typ: &Typ) -> bool {
         TypX::Air(_) => panic!("internal error: uses_ext_equal of Air"),
         TypX::StrSlice => false,
         TypX::Char => false,
-        TypX::Primitive(_, _) => true,
+        TypX::Primitive(crate::ast::Primitive::Array, _) => true,
+        TypX::Primitive(crate::ast::Primitive::Slice, _) => true,
```

Comment:
Is this to cause a type check failure if we add more primitives?

Reply:
Yes, in case there's a new `Primitive` for which `uses_ext_equal` should return a different result.